posts/2023-12-12-23.0.0.12.adoc

```diff
@@ -0,0 +1,530 @@
+name: my-ingress
+spec:
+rules:
+- http:
+    paths:
+    - path: /appA
+        pathType: Prefix
+        backend:
+        service:
+            name: appA
+            port:
+            number: 9080
+----
+
+You might use the following `server.xml` configuration to ensure that the OpenAPI UI is available at `/appA/openapi/ui`:
+
+[source,xml]
+----
+<mpOpenAPI docPath="/appA/openapi" />
+----
+
+When `uiPath` is not set, it defaults to the value of `docPath` with `/ui` appended.
+
+For more information about MicroProfile OpenAPI, see:
+
+* link:https://github.com/eclipse/microprofile-open-api[MicroProfile OpenAPI repo]
+* link:{url-prefix}/docs/latest/documentation-openapi.html[API documentation with OpenAPI]
+
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/27048
+// Contact/Reviewer: Zech-Hein
+// // // // // // // //
+
+// The following excerpt for issue https://github.com/OpenLiberty/open-liberty/issues/26138 was found in 2023-09-26-23.0.0.10-beta.adoc.
+// ------ <Excerpt From Previous Post: Start> ------
+// Contact/Reviewer: Zech-Hein
+// // // // // // // //
+
+[#ltpa]
+== Support LTPA keys rotation without a planned outage
+
+Open Liberty can now automatically generate new primary LTPA keys files while it continues to use validation keys files to validate LTPA tokens. With this update, you can rotate LTPA keys without any disruption to the application user experience. Previously, application users had to log in to their applications again after the Liberty server LTPA keys were rotated, which is no longer necessary.
+
+Primary Keys are LTPA keys in the specified keys default `ltpa.keys` file. Primary keys are used both for generating new LTPA tokens and for validating LTPA tokens. One one primary keys file is permitted per Liberty runtime.
+
+Validation keys are LTPA keys in any `.keys` files other than the primary keys file. The validation keys are used only for validating LTPA tokens. They are not used for generating new LTPA tokens. All validation keys must be located in the same directory as the primary keys file.
+
+Two ways are available to enable LTPA keys rotation without a planned outage: monitoring the primary keys file directory or specifying the validation keys file.
+
+Monitor the directory of the primary keys file for any new validation keys files::
++
+Enable the `monitorValidationKeysDir` and `monitorInterval` attributes. For example, add the following configurations to the server.xml:
++
+[source,xml]
+----
+<ltpa monitorValidationKeysDir="true" monitorInterval="5s"/>
+----
++
+The `monitorValidationKeysDir` attribute monitors the directory of the primary keys file for any `.keys` extension files. By default, this directory is `${server.config.dir}/resources/security/` but it can be configured. The Open Liberty server loads these LTPA keys and uses them as validation keys.
```

Comment:
```suggestion
The `monitorValidationKeysDir` attribute monitors the directory of the primary keys file for any `.keys` extension files. By default, this directory is `${server.config.dir}/resources/security/` but it can be configured. The Liberty server loads these LTPA keys and uses them as validation keys.
```